bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/CloudViewerImpl.java

```diff
@@ -0,0 +1,209 @@
+    private String defaultViewMode;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String viewerHeight;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean borderless;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showAnnotationTools;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showFullScreen;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showLeftHandPanel;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showDownloadPdf;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showPrintPdf;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean showPageControls;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private boolean dockPageControls;
+
+    @Inject
+    private Resource resource;
+
+    @Inject
+    @Optional
+    private CloudViewerConfigService config;
+
+    @Override
+    public String getClientId() {
+        return config.getClientId();
+    }
+
+    @Override
+    public String getReportSuiteId() {
+        return config.getReportSuiteId();
+    }
+
+    @Override
+    public String getDocumentPath() {
+        return documentPath;
+    }
+
+    @Override 
+    public String getDocumentFileName() {
+        int index = documentPath.lastIndexOf("/");
+        return documentPath.substring(index + 1);
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    public String getDefaultViewMode() {
+        return defaultViewMode;
+    }
+
+    @Override
+    public String getViewerHeight() {
+        return viewerHeight;
+    }
+
+    @Override
+    public boolean getBorderless() {
+        return borderless;
+    }
+
+    @Override
+    public boolean getShowAnnotationTools() {
+        return showAnnotationTools;
+    }
+
+    @Override
+    public boolean getShowFullScreen() {
+        return showFullScreen;
+    }
+
+    @Override
+    public boolean getShowLeftHandPanel() {
+        return showLeftHandPanel;
+    }
+
+    @Override
+    public boolean getShowDownloadPdf() {
+        return showDownloadPdf;
+    }
+
+    @Override
+    public boolean getShowPrintPdf() {
+        return showPrintPdf;
+    }
+
+    @Override
+    public boolean getShowPageControls() {
+        return showPageControls;
+    }
+
+    @Override
+    public boolean getDockPageControls() {
+        return dockPageControls;
+    }
+
+    @Override
+    public String getViewerConfigJson() {
+        JsonObjectBuilder jsonObjectBuilder = Json.createObjectBuilder();
+        jsonObjectBuilder.add("embedMode", type);
+
+        if (type.equals("FULL_WINDOW")) {
+            jsonObjectBuilder.add("defaultViewMode", defaultViewMode);
```

Comment:
I would rather we define constants for property names, as part of our interface/API and use those instead of magic strings.

Reply:
This was not addressed, see: `defaultViewMode `, `showAnnotationTools `, `showLeftHandPanel `...